core/src/main/java/com/datastax/oss/driver/internal/core/os/JnrNativeImpl.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.internal.core.os;
+
+import java.util.Optional;
+import jnr.posix.POSIX;
+import jnr.posix.POSIXFactory;
+import jnr.posix.Timeval;
+import jnr.posix.util.DefaultPOSIXHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class JnrNativeImpl extends AbstractNativeImpl {
+
+  private static final Logger LOG = LoggerFactory.getLogger(JnrNativeImpl.class);
+
+  private final Optional<POSIX> posix;
```

Comment:
@adutra I'll speak to the functional style point shortly, but I'm curious on the first point... what specifically is your IDE complaining about re: storing an Optional type as a field?  This is... pretty common.  Is it a question of naming or complaints about the types?

Re: serializable classes... the concern there in a Spark context has to do with DAOs (and their equivalents) being serializable.  There's no need for every class in the driver to be serializable, and there's no call (that I've heard of) to make Native serializable.  The old version certainly was not so it's not clear to me why we'd need to change it here.

Reply:
No strong feeling on functional or not (I find that optional often falls short in Java, but that's entirely personal opinion).

With regard to serializability, I don't think this will be an issue since the only instance of this class is a static field in a utility class.

Reply:
Re: the functional question... I readily agree this is largely a matter of style.  That said, my implementation included an attempt to be more explicit about what was going on, specifically around the loading vs. validation steps.  If we collapse those operations (as you do in your implementation @adutra) I'd change my version a fair amount as well.

It's also worth noting that you're still doing several checks along the lines of "if this val is null return some an empty option, otherwise return a non-empty option".  This indicates that you aren't fully leveraging the notion of the optional type.  I also note that you have several repeated try/catch and blocks which follow a very similar pattern but wind up contributing to code cruft.  The more functional/optional-friendly version allows us to filter those out.  My take on this (see https://github.com/datastax/java-driver/blob/52573a54ccc896c912e67640cd0679af062624e2/core/src/main/java/com/datastax/oss/driver/internal/core/os/JnrNativeImpl.java) is (in my opinion) quite a bit more concise than the verbose Java example above.

Again, though, I readily agree that this question is largely a matter of taste/style.